### PR TITLE
[feat][cli] Add read command to pulsar-client-tools

### PR DESCRIPTION
**Pull request by massakam**
### Motivation

I thought it would be convenient to create a reader easily with CLI, so I added the read command to pulsar-client-tools.

Example of use:
```sh
$ ./bin/pulsar-client read -m 48:-1 -n 5 persistent://public/default/t1

----- got message -----
key:[null], properties:[], content:m0
----- got message -----
key:[null], properties:[], content:m1
----- got message -----
key:[null], properties:[], content:m2
----- got message -----
key:[null], properties:[], content:m3
----- got message -----
key:[null], properties:[], content:m4
```

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Documentation

- [x] `doc-required`

### CODE REVIEW

**Review by nicoloboschi — COMMENTED** [review]
I do support a lot this pull request, this will help understand what it's in a topic easily

I was thinking a while ago to implement the same tool and some ideas (for future improvements)
- dump the messages to a file in a JSON-lines format
- filter out messages by message properties

I'm not saying we have to do it now, just some ideas for future enhancements

Have you tested it with pulsar-shell?

**Comment by massakam** [comment]
> Have you tested it with pulsar-shell?

It seems to work fine on pulsar-shell as well.
```sh
$ ./bin/pulsar-shell
Using directory: /home/massakam/.pulsar-shell
Welcome to Pulsar shell!
  Service URL: pulsar://localhost:6650/
  Admin URL: http://localhost:8080/

Type help to get started or try the autocompletion (TAB button).
Type exit or quit to end the shell session.

default(localhost)> client read -m Earliest -n 5 persistent://public/default/t1
2023-01-20T23:25:24,112+0900 [pulsar-client-io-71-3] INFO  org.apache.pulsar.client.impl.ConnectionPool - [[id: 0x05f137b5, L:/127.0.0.1:43390 - R:localhost/127.0.0.1:6650]] Connected to server
2023-01-20T23:25:24,118+0900 [pulsar-client-io-71-3] INFO  org.apache.pulsar.client.impl.ConsumerStatsRecorderImpl - Starting Pulsar consumer status recorder with config: {"topicNames":["persistent://public/default/t1"],"topicsPattern":null,"subscriptionName":"reader-749d586940","subscriptionType":"Exclusive","subscriptionProperties":null,"subscriptionMode":"NonDurable","receiverQueueSize":1000,"acknowledgementsGroupTimeMicros":100000,"maxAcknowledgmentGroupSize":1000,"negativeAckRedeliveryDelayMicros":60000000,"maxTotalReceiverQueueSizeAcrossPartitions":50000,"consumerName":null,"ackTimeoutMillis":0,"tickDurationMillis":1000,"priorityLevel":0,"maxPendingChunkedMessage":10,"autoAckOldestChunkedMessageOnQueueFull":false,"expireTimeOfIncompleteChunkedMessageMillis":60000,"cryptoFailureAction":"FAIL","properties":{},"readCompacted":false,"subscriptionInitialPosition":"Latest","patternAutoDiscoveryPeriod":60,"regexSubscriptionMode":"PersistentOnly","deadLetterPolicy":null,"retryEnable":false,"autoUpdatePartitions":true,"autoUpdatePartitionsIntervalSeconds":60,"replicateSubscriptionState":false,"resetIncludeHead":false,"batchIndexAckEnabled":false,"ackReceiptEnabled":false,"poolMessages":true,"startPaused":false,"autoScaledReceiverQueueSizeEnabled":false,"topicConfigurations":[],"maxPendingChuckedMessage":10}
2023-01-20T23:25:24,119+0900 [pulsar-client-io-71-3] INFO  org.apache.pulsar.client.impl.ConsumerStatsRecorderImpl - Pulsar client config: {"serviceUrl":"pulsar://localhost:6650/","authPluginClassName":null,"authParams":null,"authParamMap":null,"operationTimeoutMs":30000,"lookupTimeoutMs":30000,"statsIntervalSeconds":60,"numIoThreads":6,"numListenerThreads":6,"connectionsPerBroker":1,"connectionMaxIdleSeconds":180,"useTcpNoDelay":true,"useTls":false,"tlsKeyFilePath":"","tlsCertificateFilePath":"","tlsTrustCertsFilePath":"","tlsAllowInsecureConnection":false,"tlsHostnameVerificationEnable":false,"concurrentLookupRequest":5000,"maxLookupRequest":50000,"maxLookupRedirects":20,"maxNumberOfRejectedRequestPerConnection":50,"keepAliveIntervalSeconds":30,"connectionTimeoutMs":10000,"requestTimeoutMs":60000,"readTimeoutMs":60000,"autoCertRefreshSeconds":300,"initialBackoffIntervalNanos":100000000,"maxBackoffIntervalNanos":60000000000,"enableBusyWait":false,"listenerName":null,"useKeyStoreTls":false,"sslProvider":null,"tlsKeyStoreType":"JKS","tlsKeyStorePath":"","tlsKeyStorePassword":"*****","tlsTrustStoreType":"JKS","tlsTrustStorePath":"","tlsTrustStorePassword":"*****","tlsCiphers":[],"tlsProtocols":[],"memoryLimitBytes":0,"proxyServiceUrl":null,"proxyProtocol":null,"enableTransaction":false,"dnsLookupBindAddress":null,"dnsLookupBindPort":0,"socks5ProxyAddress":null,"socks5ProxyUsername":null,"socks5ProxyPassword":null}
2023-01-20T23:25:24,121+0900 [pulsar-client-io-71-3] INFO  org.apache.pulsar.client.impl.ConsumerImpl - [persistent://public/default/t1][reader-749d586940] Subscribing to topic on cnx [id: 0x05f137b5, L:/127.0.0.1:43390 - R:localhost/127.0.0.1:6650], consumerId 0
2023-01-20T23:25:24,124+0900 [pulsar-client-io-71-3] INFO  org.apache.pulsar.client.impl.ConsumerImpl - [persistent://public/default/t1][reader-749d586940] Subscribed to topic on localhost/127.0.0.1:6650 -- consumer: 0
----- got message -----
key:[null], properties:[], content:test
----- got message -----
key:[null], properties:[], content:test
----- got message -----
key:[null], properties:[], content:test
----- got message -----
key:[null], properties:[], content:test
----- got message -----
key:[null], properties:[], content:test
2023-01-20T23:25:24,135+0900 [pulsar-client-io-71-3] INFO  org.apache.pulsar.client.impl.ConsumerImpl - [persistent://public/default/t1] [reader-749d586940] Closed consumer
2023-01-20T23:25:24,137+0900 [main] INFO  org.apache.pulsar.client.impl.PulsarClientImpl - Client closing. URL: pulsar://localhost:6650/
2023-01-20T23:25:24,137+0900 [pulsar-client-io-71-3] INFO  org.apache.pulsar.client.impl.ClientCnx - [id: 0x05f137b5, L:/127.0.0.1:43390 ! R:localhost/127.0.0.1:6650] Disconnected
2023-01-20T23:25:26,141+0900 [main] INFO  org.apache.pulsar.client.cli.PulsarClientTool - 5 messages successfully read
default(localhost)>

```

File: pulsar-client-tools/src/test/java/org/apache/pulsar/client/cli/TestCmdRead.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        assertEquals(cmdRead.getWebSocketReadUri(topicNameV1),
+                "ws://localhost:8080/ws/reader/persistent/public/cluster/default/t1?messageId=" + msgIdQueryParam);
+
+        String topicNameV2 = "persistent://public/default/t2";
+        assertEquals(cmdRead.getWebSocketReadUri(topicNameV2),
+                "ws://localhost:8080/ws/v2/reader/persistent/public/default/t2?messageId=" + msgIdQueryParam);
+    }
+
+    @Test
+    public void testPrseMessageId() {
```
Comment (nkurihar) [inline pulsar-client-tools/src/test/java/org/apache/pulsar/client/cli/TestCmdRead.java:61]: maybe typo:
```suggestion
    public void testParseMessageId() {
```

Comment (massakam) [reply pulsar-client-tools/src/test/java/org/apache/pulsar/client/cli/TestCmdRead.java]: Fixed.

**Review by nkurihar — APPROVED** [review]
LGTM

**Comment by momo-jun** [comment]
Hi @massakam, It will help users a lot if you can add the docs for this feature. Do you have any planned updates on that?

**Comment by massakam** [comment]
@momo-jun Yes, I will make a pull-request to the pulsar-site repository later.

**Comment by massakam** [comment]
@momo-jun The reference for the "read" subcommand added in this PR seems to have been automatically generated and committed to the pulsar-site repository. So I don't think I should do more.
https://github.com/apache/pulsar-site/commit/75f16503bc3ce96e710f5072234b1178b3a8aff8

**Pull request merged**